pages/clubs/Running.jsx

```diff
@@ -9,39 +11,29 @@ var Running = React.createClass({
   contextTypes: {
     intl: React.PropTypes.object
   },
-  setData : function(data){
-    for(var i = 0; i < data.length; i++){
-      var categoryLabel = data[i].category.toLowerCase();
+  componentDidMount: function() {
+    this.props.teachAPI.getClubsGuides((err, res) => {
+      if(err) {
+        this.setState({error: err});
+        return;
+      }
 
-      categoryLabel = "guide_category_" + categoryLabel.split(' ').join('_');
+      var data = res.body;
 
-      var localizedCategory = this.context.intl.formatMessage({id: categoryLabel});
+      data.forEach(dataItem => {
+        var categoryLabel = dataItem.category.toLowerCase();
 
-      if(localizedCategory !== categoryLabel){
-        data[i].category = localizedCategory;
-      }
-    }
-    this.setState({guides : data });
-  },
-  fetchJSON : function(path,callback){
-    var httpRequest = new XMLHttpRequest();
+        categoryLabel = "guide_category_" + categoryLabel.split(' ').join('_');
```

Comment:
this seems extremely fragile, where is this label format coming from? (can we somehow tap into whatever dictates this format?)